doc/api/querystring.md

```diff
@@ -4,81 +4,140 @@
 <!-- YAML
 added: v0.1.25
 -->
 
-Serialize an object to a query string.
-Optionally override the default separator (`'&'`) and assignment (`'='`)
-characters.
+* `obj` {Object} The object to serialize into a URL query string
+* `sep` {String} The substring used to delimit key+value pairs in the
+  query string. Defaults to `'&'`.
+* `eq` {String}. The substring used to delimit keys and values in the
+  query string. Defaults to `'='`.
+* `options`
+  * `encodeURIComponent` {Function} The function is use when encoding
+    percent-encoded characters in the query string. Defaults to
+    `querystring.escape()`.
+
+The `querystring.stringify()` method produces a URL query string from a
+given `object` by iterating through the objects "own properties".
 
-Options object may contain `encodeURIComponent` property (`querystring.escape` by default),
-it can be used to encode string with `non-utf8` encoding if necessary.
+By default, the ASCII `&` character is used to separate pairs of key and value
+pairs in the string, and the ASCII `=` character is used to separate the keys
+from the values. Alternative delimiters may be specified using the `sep` and
+`eq` arguments.
 
-Example:
+For example:
 
 ```js
 querystring.stringify({ foo: 'bar', baz: ['qux', 'quux'], corge: '' })
 // returns 'foo=bar&baz=qux&baz=quux&corge='
 
 querystring.stringify({foo: 'bar', baz: 'qux'}, ';', ':')
 // returns 'foo:bar;baz:qux'
+```
+
+By default, characters requiring percent-encoding within the query string will
+be encoded as UTF-8. If an alternative encoding is required, then an alternative
+`encodeURIComponent` option will need to be specified as illustrated in the
+following example:
+
+```js
+// Assuming gbkEncodeURIComponent function already exists,
 
-// Suppose gbkEncodeURIComponent function already exists,
-// it can encode string with `gbk` encoding
 querystring.stringify({ w: '中文', foo: 'bar' }, null, null,
   { encodeURIComponent: gbkEncodeURIComponent })
-// returns 'w=%D6%D0%CE%C4&foo=bar'
 ```
 
-## querystring.unescape
+## querystring.unescape(str)
 <!-- YAML
 added: v0.1.25
 -->
+* `str` {String}
+
+
+The `querystring.unescape()` method performs decoding of URL percent-encoded
+characters on the given `str`.
 
-The unescape function used by `querystring.parse`,
-provided so that it could be overridden if necessary.
+The `querystring.unescape()` method is used by `querystring.parse()` and is
+generally not expected to be used directly. It is exported primarily to allow
+application code to provide a replacement decoding implementation if
+necessary by assigning `querystring.unescape` to an alternative function.
 
-It will try to use `decodeURIComponent` in the first place,
-but if that fails it falls back to a safer equivalent that
-doesn't throw on malformed URLs.
+By default, the `querystring.unescape()` method will attempt to use the
+JavaScript built-in `decodeURIComponent()` method to decode. If that fails,
```

Comment:
@benjamingr It is a [built-in JavaScript function](http://www.ecma-international.org/ecma-262/6.0/#sec-decodeuricomponent-encodeduricomponent).